### PR TITLE
feat: Add "CREATE DYNAMIC TABLE" to schema_grant

### PR DESCRIPTION
**Pull request by kenkoooo**
Add `CREATE DYNAMIC TABLE` to schema_grant to allow users to create a schema with this privilege.

## References

* #1922

### CODE REVIEW

**Comment by kenkoooo** [comment]
@sfc-gh-asawicki
Hi, if I missed something I needed to do before creating a PR, please let me know! Thank you!

**Comment by kenkoooo** [comment]
@sfc-gh-asawicki @sfc-gh-jcieslak @sfc-gh-swinkler 
Hi, would you mind reviewing this PR or letting me know if I missed anything required for requesting reviews? Thank you!

**Review by sfc-gh-swinkler — DISMISSED** [review]

File: pkg/resources/schema_grant.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ import (
 
 var validSchemaPrivileges = NewPrivilegeSet(
 	privilegeAddSearchOptimization,
+	privilegeCreateDynamicTable,
```
Comment (sfc-gh-asawicki) [inline pkg/resources/schema_grant.go:19]: For completeness, it should also be added to the `pkg/sdk/privileges.go` file (`SchemaPrivilege` type).

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

**Comment by kenkoooo** [comment]
@sfc-gh-asawicki @sfc-gh-swinkler 
Thank you for your review! I have added it to the file. Would you mind reviewing it again?

**Review by sfc-gh-asawicki — APPROVED** [review]

**Comment by kenkoooo** [comment]
Thank you guys for working on it!

**Pull request merged**